src/main/java/org/myrobotlab/programab/Session.java

```diff
@@ -122,7 +124,12 @@ public void savePredicates() {
    */
   public Map<String, String> getPredicates() {
     TreeMap<String, String> sort = new TreeMap<>();
-    sort.putAll(getChat().predicates);
+    // copy keys, making this sort thread safe

```

Comment:
if there's a thread safety concern here, it should be synchronized.. Did you see issues with thread safety here?

Reply:
its a thread safety issue around maps and iterations of the map, not really about the function, but yeah I'll synchronize it